CMakeLists.txt

```diff
@@ -1,12 +1,23 @@
 CMAKE_MINIMUM_REQUIRED(VERSION 3.11)
 
+SET(CMAKE_PROJECT_INCLUDE_BEFORE
+	"${CMAKE_SOURCE_DIR}/cmake/IncludeBeforeProject.cmake"
+)
+
+PROJECT(gemrb)
+
```

Comment:
why did you move this around?

Reply:
1. Note mentioned here: https://cmake.org/cmake/help/latest/command/project.html#usage
2. `CMAKE_SYSTEM_NAME` is empty before `project` call so i couldn't set this properly:

```
IF(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
	LIST(APPEND CMAKE_MODULE_PATH
		"${CMAKE_CURRENT_SOURCE_DIR}/platforms/apple/cmake")
ENDIF()
```

Reply:
Your own CACHE_BUILD_TYPE says it should happen before the project call, so it needs to be moved if so, now that it's ran later.

I see now that CMAKE_PROJECT_INCLUDE_BEFORE requires 3.15, so please update the minimum version as well.